File: 897-increasing-order-search-tree.java
class Solution {
    TreeNode ans;
    public TreeNode increasingBST(TreeNode root) {
        TreeNode dummy = new TreeNode();
        ans = dummy;
        dfs(root);
        return dummy.right;
    }
    private void dfs(TreeNode node){
        if(node == null) return;
        dfs(node.left);
        ans.right = new TreeNode(node.val);
        ans = ans.right;
        dfs(node.right);
    }
}

File: 897-increasing-order-search-tree.py
class Solution:
    def increasingBST(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        dummy = TreeNode(0)
        ans = dummy
        def dfs(node):
            nonlocal ans
            if not node :
                return
            dfs(node.left)
            ans.right = TreeNode(node.val)
            ans = ans.right 
            dfs(node.right)
        dfs(root)
        return dummy.right
